admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorer/template.vue

```diff
@@ -653,27 +648,34 @@ export default {
                 }
             },
 
-            deletePage: function(me, target) {
-                $perAdminApp.askUser('Delete Page', me.$i18n('Are you sure you want to delete this node and all its children?'), {
-                    yes() {
-                        const resourceType = target.resourceType
-                        if(resourceType === 'per:Object') {
-                            $perAdminApp.stateAction('deleteObject', target.path)
-                        } else if(resourceType === 'per:Asset') {
-                            $perAdminApp.stateAction('deleteAsset', target.path)
-                        } else if(resourceType === 'sling:OrderedFolder') {
-                            $perAdminApp.stateAction('deleteFolder', target.path)
-                        } else if(resourceType === 'per:Page') {
-                            $perAdminApp.stateAction('deletePage', target.path)
-                        } else if(resourceType === 'nt:file') {
-                            $perAdminApp.stateAction('deleteFile', target.path)
-                        }else {
-                            $perAdminApp.stateAction('deleteFolder', target.path)
+            handleDelete: function(type, path) {
+                const me = this
+                return new Promise((resolve, reject) => {
+                    $perAdminApp.askUser(`Delete ${type}?`, me.$i18n(`Are you sure you want to delete this node and all its children?`), {
+                        yes() {
+                            $perAdminApp.stateAction(`delete${type.charAt(0).toUpperCase() + type.slice(1)}`, path)
+                            resolve()
                         }
-                    }
+                    })
                 })
             },
 
+            deletePage: function(me, target) {
+                const { resourceType, path } = target
+                let action = 'folder'
```

Comment:
I thinks it's a bit confusing here that `action` is the name of the variable when it is passed as `type` to `handleDelete()`

Reply:
Thanks, good catch!